contentcuration/contentcuration/frontend/settings/views/UsingStudio/ReportIssueForm.vue

```diff
@@ -0,0 +1,136 @@
+<template>
+  
+  <form @submit.prevent="submit">
+    <div style="padding-bottom: 16px;">
+      <KTextbox
+        v-model='operating_system'
+        label="Operating system (e.g. Windows, MacOS, Linux)"
+        :invalid="errors.includes('operating_system')"
+        invalidText="This field is required"
+      />
+      <KTextbox
+        v-model='browser'
+        label="Browser (e.g. Chrome, Firefox, Safari)"
+        :invalid="errors.includes('browser')"
+        invalidText="This field is required"
+      />
+      <KTextbox
+        v-model='channel'
+        label="Channel you discovered the issue in"
+        :invalid="errors.includes('channel')"
+        invalidText="This field is required"
+      />
+      <KTextbox
+        v-model='description'
+        label="Describe your issue with as much detail as possible"
+        :textArea="true"
+        :invalid="errors.includes('description')"
+        invalidText="This field is required"
+      />
+    </div>
+    <KButtonGroup style="text-align: right; display: block;">
+      <KButton appearance="flat-button" text="Cancel" @click="$emit('close')" />
+      <KButton :primary="true" text="Submit" @click="submit" />
+    </KButtonGroup>
+  </form>
+
+</template>
+
+
+<script>
+
+  import client from "../../../shared/client";
+
+  export default {
+    name: 'ReportIssueForm',
+    data() {
+      return {
+        operating_system: '',
+        browser: '',
+        channel: '',
+        description: '',
+        errors: [],
+      };
+    },        
+    methods: {
+      submit() {
+        const formData = {
+          operating_system: this.operating_system,
+          browser: this.browser,
+          channel: this.channel,
+          description: this.description,
+          errors: [],
+        };
+
+        if(!this.validate(formData)) {
+          return;
+        }
+
+        const path = window.Urls.issues_settings();
+
+        client.post(path, formData)
+          .then(response => {
+            this.$store.dispatch('showSnackbar', { text: response.data });
+          })
+          .catch(e => {
+            this.$store.dispatch('showSnackbar', { text: "Failed to submit the form" });
+          })
+          .finally(() => this.$emit('close'));
+      },
+      validate(formData) {
+        let errorKeys = []
+        Object.keys(formData).forEach(k => {
+          if(Boolean(formData[k])) {
+            return;
+          } else {
+            if(this.errors.includes(k)) {
+              return;
+            } else {
+              this.errors.push(k);
+            }
+          }
+        });
+      },
+    },
+    watch: {
+      operating_system(val) {
```

Comment:
Since it looks like these all function pretty much the same, would it make sense to move this to computed props and consolidating the logic like [here](https://github.com/learningequality/studio/blob/vue-refactor/contentcuration/contentcuration/static/js/edit_channel/uploader/views/DetailsTabView.vue#L296)?